src/controllers/asteroid-controller.js

```diff
@@ -0,0 +1,21 @@
+const { mapQueryToMeteorRequest } = require('../utils');
+const { asteroidService } = require('../services');
+
+const handleError = (res, error) => {
+    res.status(500).send(error.message);
+}
+
+const getAsteroids = async (req, res) => {
+    const request = mapQueryToMeteorRequest(req.query);
+
+    try {
+        const data = await asteroidService.getAsteroidsWithinPeriod(request);
+        res.json({ data });
+    } catch (error) {
+        (error) => handleError(res, error);
```

Comment:
```js
 } catch (error) {
  next(error)
}
```

end then you can handle this error in the next middleware

Reply:
Updated.